src/main/java/io/github/opencubicchunks/cubicchunks/chunk/ticket/PlayerCubeTicketTracker.java

```diff
@@ -11,81 +12,127 @@
+                            if (this.isWithinViewDistance(horizontalGraphGroup.getLevel(pos), verticalGraphGroup.getLevel(pos))) {
+                                iTicketManager.registerCube(pos, ticket);
+                                iTicketManager.getCubePositions().add(pos);
                             } else {
                                 iTicketManager.getPlayerCubeTicketThrottlerSorter().enqueue(CubeTaskPriorityQueueSorter.createSorterMsg(() -> {
-                                }, cubePosIn, false));
+                                }, pos, false));
                             }
 
-                        }), cubePosIn, () -> distance));
+                        }), pos, () -> horizDistance));
             } else {
                 iTicketManager.getPlayerCubeTicketThrottlerSorter().enqueue(CubeTaskPriorityQueueSorter.createSorterMsg(() ->
-                        iTicketManager.executor().execute(() ->
-                                iTicketManager.releaseCube(cubePosIn, ticket)),
-                        cubePosIn, true));
+                                iTicketManager.executor().execute(() ->
+                                        iTicketManager.releaseCube(pos, ticket)),
+                        pos, true));
             }
         }
 
     }
 
+    /**
+     * Deal with all of the updates to the graphs
+     */
     public void processAllUpdates() {
-        super.processAllUpdates();
+        horizontalGraphGroup.processAllUpdates();
+        verticalGraphGroup.processAllUpdates();
         if (!this.positionsAffected.isEmpty()) {
-            LongIterator longiterator = this.positionsAffected.iterator();
+            LongIterator itrPositions = this.positionsAffected.iterator();
+
+            while (itrPositions.hasNext()) {
+                long pos = itrPositions.nextLong();
 
-            while (longiterator.hasNext()) {
-                long i = longiterator.nextLong();
-                int j = this.distances.get(i);
-                int k = this.getLevel(i);
-                if (j != k) {
+                int oldHorizDistance = this.horizDistances.get(pos);
+                int oldVertDistance = this.vertDistances.get(pos);
+                int horizCurrentDistance = horizontalGraphGroup.getLevel(pos);
+                int vertCurrentDistance = verticalGraphGroup.getLevel(pos);
+
+                if (oldHorizDistance != horizCurrentDistance || oldVertDistance != vertCurrentDistance) {
                     //func_219066_a = update level
-                    iTicketManager.getCubeTaskPriorityQueueSorter().onUpdateCubeLevel(CubePos.from(i), () -> this.distances.get(i), k, (ix) -> {
-                        if (ix >= this.distances.defaultReturnValue()) {
-                            this.distances.remove(i);
+                    iTicketManager.getCubeTaskPriorityQueueSorter().onUpdateCubeLevel(CubePos.from(pos), () -> this.horizDistances.get(pos), horizCurrentDistance, (horizDistance) -> {
```

Comment:
Take vertical view distance into account when sorting cube load tasks
```suggestion
                    iTicketManager.getCubeTaskPriorityQueueSorter().onUpdateCubeLevel(CubePos.from(pos), () -> this.horizDistances.get(pos), horizCurrentDistance+verticalViewDistance, (horizDistance) -> {
```